src/PolyhedralGeometry/PolyhedralFan/properties.jl

```diff
@@ -281,6 +281,49 @@ function cones(PF::_FanLikeType)
   return IncidenceMatrix([Vector{Int}(x) for x in cones])
 end
 
+@doc raw"""
+    minimal_supercone(X::NormalToricVariety, p::AbstractVector{<:IntegerUnion})
+    -> SubObjectIterator{RayVector}
+
+Given an point $p$ with integer coordinates inside the support of the fan $Σ$ of a normal toric variety~$X$, return the unique cone $σ$ in $Σ$ such that $p$ is in the relative interior of $σ$.
+
+# Examples
+```jldoctest
+julia> X = projective_space(NormalToricVariety, 3)
+Normal toric variety
+
+julia> p = [1, 1, 0]
+3-element Vector{Int64}:
+ 1
+ 1
+ 0
+
+julia> c = minimal_supercone(X, p)
+Polyhedral cone in ambient dimension 3
+
+julia> rays(c)
+2-element SubObjectIterator{RayVector{QQFieldElem}}:
+ [0, 1, 0]
+ [1, 0, 0]
+```
+"""
+function minimal_supercone(
+  X::NormalToricVariety, r::AbstractVector{<:Union{IntegerUnion,QQFieldElem}}
+)
+  m_cone_indices = Oscar._get_maximal_cones_containing_vector(X, r)
+  @req !is_empty(m_cone_indices) "Ray `r` should be in the support of the fan of `X`."
+  m_cones = [maximal_cones(X)[i] for i in m_cone_indices]
+  m_cone = intersect(m_cones)
+  m_fan = polyhedral_fan(m_cone)
+
+  # By Proposition 1.2.8(c) in [CLS11](@cite), every proper face of a cone
+  # is the intersection of the facets containing it.
+  m_facets = cones(m_fan, dim(m_cone) - 1)
+  cs = [c for c in m_facets if r in c]
+  isempty(cs) && return m_cone
+  return intersect(cs)
+end
```

Comment:
This is too expensive. Why are you intersecting cones that belong to the same fan? You should intersect the corresponding index sets. Why construct a fan out of `m_cone`? The intersection of cones `m_cone` is already the minimal cone containing `r`. I am lost for the remainder of this function. My suggestion is the following:

```suggestion
function minimal_supercone(
  X::NormalToricVariety, r::AbstractVector{<:Union{IntegerUnion,QQFieldElem}}
)
  m_cone_indices = Oscar._get_maximal_cones_containing_vector(X, r)
  @req !is_empty(m_cone_indices) "Ray `r` should be in the support of the fan of `X`."
  IX = maximal_cones(IncidenceMatrix, X)
  m_cone = intersect([row(IX, i) for i in m_cone_indices]...)
  RX = matrix(coefficient_field(X), rays(X))
  return positive_hull(RX[[m_cone...], :], lineality_space(X))
end
```

Probably @benlorenz has further improvements.

Reply:
> The intersection of cones m_cone is already the minimal cone containing r

If the fan has one maximal cone, then `m_cone` is just that one cone, which is not necessarily the minimal cone in the fan containing the point. In the attached picture, the ray with minimal generator `(1, 0, 1)` is in the relative interior of a two-dimensional cone in the fan. That is the cone that this function returns.

![Screenshot from 2024-12-06 15-13-45](https://github.com/user-attachments/assets/521ddb70-e48d-4206-b41d-991b69d3e162)

Reply:
It would be good if we can only work with indices and perhaps just return the index of the cone. I was not sure how to do this

Reply:
You're right. My claim is even wrong when there are more maximal cones. I apologize for that.

Reply:
I will work something out using the Hasse diagram.

Reply:
In the end this is the same as your approach, but I managed to avoid expensive intersections:
```
julia> m_cone = positive_hull([1 0 0; 0 1 0; 0 0 1])
Polyhedral cone in ambient dimension 3

julia> r = [1,1, 0]
3-element Vector{Int64}:
 1
 1
 0

julia> r in m_cone
true

julia> Fm = facets(m_cone)
3-element SubObjectIterator{LinearHalfspace{QQFieldElem}} over the halfspaces of R^3 described by:
-x_1 <= 0
-x_2 <= 0
-x_3 <= 0


julia> zero_facets = findall(f -> f.a*r == [0], Fm)
1-element Vector{Int64}:
 3

julia> Rm = matrix(coefficient_field(m_cone), rays(m_cone))
[1   0   0]
[0   1   0]
[0   0   1]

julia> RIF = IncidenceMatrix(m_cone.pm_cone.RAYS_IN_FACETS)
3×3 IncidenceMatrix
[2, 3]
[1, 3]
[1, 2]


julia> result_ray_indices = intersect([row(RIF, i) for i in zero_facets]...)
Set{Int64} with 2 elements:
  2
  1

julia> result = positive_hull(Rm[[result_ray_indices...], :], lineality_space(m_cone))
Polyhedral cone in ambient dimension 3
```
So the general thing could look like:
```suggestion
function minimal_supercone(
  X::NormalToricVariety, r::AbstractVector{<:Union{IntegerUnion,QQFieldElem}}
)
  m_cone_indices = Oscar._get_maximal_cones_containing_vector(X, r)
  @req !is_empty(m_cone_indices) "Ray `r` should be in the support of the fan of `X`."
  IX = maximal_cones(IncidenceMatrix, X)
  m_cone = intersect([row(IX, i) for i in m_cone_indices]...)
  RX = matrix(coefficient_field(X), rays(X))
  m_cone = positive_hull(RX[[m_cone...], :], lineality_space(X))
  Fm = facets(m_cone)
  zero_facets = findall(f -> f.a*r == [0], Fm)
  Rm = matrix(coefficient_field(m_cone), rays(m_cone))
  RIF = IncidenceMatrix(m_cone.pm_cone.RAYS_IN_FACETS)
  result_ray_indices = intersect([row(RIF, i) for i in zero_facets]...)
  result = positive_hull(Rm[[result_ray_indices...], :], lineality_space(m_cone))
  return result
end
```
This avoids the expensive intersections, nevertheless it requires a convex hull computation. Might be more if the fan did not have rays before. Hope this works for you.

Reply:
The tests failed when some of the defined sets were empty. I pushed a commit that dealt with three such special cases in the above. Now the tests seem to pass, so I hope it is correct.

Returning the index of the resulting cone would also be OK if that is faster.

One side-comment that I am less familiar with polymake, so the variable names RIF, Fm, RX, IX, etc are very hard to read for me.

Reply:
Instead of tests with affine_space and projective_space, I guess the tests can be replaced by fan tests, so it can be entirely under PolyhedralGeometry